ironic/tests/unit/drivers/modules/drac/test_raid.py

```diff
@@ -1380,48 +1406,47 @@ def test_create_configuration_fails_if_disk_already_reserved(
+                       autospec=True)
+    @mock.patch.object(drac_raid, 'list_raid_controllers', autospec=True)
     @mock.patch.object(drac_job, 'validate_job_queue', spec_set=True,
                        autospec=True)
     @mock.patch.object(drac_raid, 'commit_config', spec_set=True,
                        autospec=True)
-    @mock.patch.object(drac_raid, 'delete_virtual_disk', spec_set=True,
+    @mock.patch.object(drac_raid, 'reset_raid_config', spec_set=True,
                        autospec=True)
-    def test_delete_configuration(self, mock_delete_virtual_disk,
+    def test_delete_configuration(self, mock_reset_raid_config,
                                   mock_commit_config,
                                   mock_validate_job_queue,
-                                  mock_list_virtual_disks,
+                                  mock_list_raid_controllers,
+                                  mock_get_reboot_options,
                                   mock_get_drac_client):
         mock_client = mock.Mock()
         mock_get_drac_client.return_value = mock_client
-        virtual_disk_dict = {
-            'id': 'Disk.Virtual.0:RAID.Integrated.1-1',
-            'name': 'disk 0',
-            'description': 'Virtual Disk 0 on Integrated RAID Controller 1',
-            'controller': 'RAID.Integrated.1-1',
-            'raid_level': '1',
-            'size_mb': 571776,
-            'status': 'ok',
-            'raid_status': 'online',
-            'span_depth': 1,
-            'span_length': 2,
-            'pending_operations': None,
-            'physical_disks': []}
-        mock_list_virtual_disks.return_value = [
-            test_utils.make_virtual_disk(virtual_disk_dict)]
+        raid_controller_dict = {
+            'id': 'RAID.Integrated.1-1',
+            'description': 'Integrated RAID Controller 1',
+            'manufacturer': 'DELL',
+            'model': 'PERC H710 Mini',
+            'primary_status': 'ok',
+            'firmware_version': '21.3.0-0009',
+            'bus': '1',
+            'supports_realtime': True}
+
+        mock_list_raid_controllers.return_value = [
+            test_utils.make_raid_controller(raid_controller_dict)]
         mock_commit_config.return_value = '42'
-        mock_delete_virtual_disk.return_value = {
+        mock_reset_raid_config.return_value = {
             'is_reboot_required': 'optional',
-            'commit_required': False,
             'is_commit_required': True}
 
         with task_manager.acquire(self.context, self.node.uuid,
                                   shared=False) as task:
             return_value = task.driver.raid.delete_configuration(task)
 
             mock_commit_config.assert_called_once_with(
-                task.node, raid_controller='RAID.Integrated.1-1', reboot=False,
-                realtime=True)
+                task.node, raid_controller='RAID.Integrated.1-1', reboot=True,
+                realtime=False)
```

Comment:
Please remember to revert this change after addressing the issue with `mock_get_reboot_options`.

Reply:
Updated with latest commit.